### PR TITLE
Consumes the values of MLMatrix44 one by one.

### LINKED ISSUE #1198: Loading single line MLMatrix44 reads only the first four values
If the 16 values of MLMatrix44 are provided without line breaks, only the first four values are read.
More generally, without a line break on each four values, the MLMatrix isn't correctly parsed.

This issue has introduced with commit 51e2261674910bb843df91be6210863dc3bf27fd to fix issue #913.

### PR DESCRIPTION
**Pull request by PeC-KAYO**
The first purpose here is to resolve #1198.
The second one is to be more compliant with the XML schema of MeshLabProject in cnr-isti-vclab/meshlab/docs/meshlabprojspecificationformat.xsd. Elements of a list (as a simple type of XML schema) are separated by space and new line breaks must be discarded.

The only one relevant commit is the 8db5e27.

### CODE REVIEW

**Comment by PeC-KAYO** [comment]
The only one relevant commit is the 8db5e2733ba1722c9a6e27645bd0ac72d37148ec.
It's my first opened issue and me first pull request. And also, my first real use of GitHub. All in a foreign language for me.
So, any feedback is welcome!

File: src/meshlabplugins/io_base/load_project.cpp [hunk 1]
```diff
@@ -230,19 +230,16 @@ std::vector<MeshModel*> loadMLP(
...
-									int j = 0;
-									for (const QString& value : qAsConst(values)) {
-										if (i < 4 && j < 4) {
-											md.mm()->cm.Tr[i][j] = value.toFloat();
-											j++;
-										}
+							auto * const v = md.mm()->cm.Tr.V();
+							auto const rows = tr.firstChild().nodeValue().split("\n", QString::SkipEmptyParts);
+							auto i = 0u;
+							for (auto const & row: rows) {
```
Comment (PeC-KAYO) [inline src/meshlabplugins/io_base/load_project.cpp:236]: Useless test.
```suggestion
							{
```

File: src/meshlabplugins/io_base/load_project.cpp [hunk 2]
```diff
@@ -230,19 +230,16 @@ std::vector<MeshModel*> loadMLP(
...
-									int j = 0;
-									for (const QString& value : qAsConst(values)) {
-										if (i < 4 && j < 4) {
-											md.mm()->cm.Tr[i][j] = value.toFloat();
-											j++;
-										}
+							auto * const v = md.mm()->cm.Tr.V();
+							auto const rows = tr.firstChild().nodeValue().split("\n", QString::SkipEmptyParts);
+							auto i = 0u;
+							{
```
Comment (PeC-KAYO) [inline src/meshlabplugins/io_base/load_project.cpp:236]:
```suggestion
							for (auto const & row: rows) {
```

File: src/meshlabplugins/io_base/load_project.cpp [hunk 3]
```diff
@@ -230,19 +230,16 @@ std::vector<MeshModel*> loadMLP(
...
+								if (rows.size() > 0u) {
```
Comment (PeC-KAYO) [inline src/meshlabplugins/io_base/load_project.cpp:237]:
```suggestion
								{
```

**Review by PeC-KAYO — COMMENTED** [review]
Revert previous review and add correct proposal. Sorry…

**Comment by alemuntoni** [comment]
Hi @PeC-KAYO, thanks for the PR. 
It looks good to me, since it will support both mlp having matrices with newlines and without them. I'll make some tests when the automatic builds will generate the binaries.

**Comment by alemuntoni** [comment]
Maybe would be a good idea also to remove the newline when saving mlp files, am I wrong?
https://github.com/cnr-isti-vclab/meshlab/blob/f35828b65fef4d85479d53207cb6e15d3dd7d937/src/meshlabplugins/io_base/save_project.cpp#L23

**Comment by PeC-KAYO** [comment]
> Maybe would be a good idea also to remove the newline when saving mlp files, am I wrong?
> 
> https://github.com/cnr-isti-vclab/meshlab/blob/f35828b65fef4d85479d53207cb6e15d3dd7d937/src/meshlabplugins/io_base/save_project.cpp#L23

Yes, I think it is a good idea. It's not wrong to add line breaks, but it can mislead as to their necessity.

**Comment by PeC-KAYO** [comment]
Sorry for the multiple commits, but I only just realized that they were dropping directly in that pull request.
Also, I can't manage to compile all the code on my machine (macOS) and generate the package. So I use GitHub for development phase… Any help appreciated.

**Comment by alemuntoni** [comment]
Thanks @PeC-KAYO for the contribution!

**Pull request merged**